cadc-registry/src/main/java/ca/nrc/cadc/reg/client/RegistryClient.java

```diff
@@ -232,24 +253,23 @@ private void init(URL resourceCapsURL) {
     }
 
     /**
-     * Get the capabilities object for the resource identified
-     * by resourceID.
+     * Obtain the URL of the capabilities document for the given resourceID.  Useful when the resource entries do not
+     * point to a capabilities document, but rather just a URL.
      *
-     * @param resourceID Identifies the resource.
-     * @return The associated capabilities object.
-     * @throws IOException If the capabilities could not be determined.
+     * If the given resource ID is not in the list of services located at <code>resourceCapsURL</code> then expect an
+     * <code>IllegalArgumentException</code> to be thrown.  If multiples ones are found, then expect a
+     * <code>RuntimeException</code> as the system will not select one for you.
+     *
+     * @param resourceID        URI of the resource to lookup.
+     * @return URL              Location of the document.
+     * @throws IOException      If the cache file cannot be read.
      */
-    public Capabilities getCapabilities(URI resourceID) throws IOException {
-        if (resourceID == null) {
-            String msg = "Input parameter (resourceID) should not be null";
-            throw new IllegalArgumentException(msg);
-        }
-
+    public URL getCapabilitiesURL(URI resourceID) throws IOException {
```

Comment:
In the VO registry model, this URL would be an accessURL, where the capabilities URL is the accessURL of one specific featuure of the resource (the VOSI-capabiltiies feature)... sounds convoluted because capabilties is all about self-describing services that can also have the description in a registry.

So, I think getAccessURL would be a better name for this method. Which accessURL you get depends on which "canned query" you are using (eg which URL you use to init the RegistryClient)

Reply:
Done and committed.